R/data_merge_module.R

```diff
@@ -245,16 +255,28 @@ data_merge_srv <- function(id = "merge_id",
       reactive({
         checkmate::assert_list(selector_list(), names = "named", types = "reactive")
         merge_fun_name <- if (inherits(merge_function, "reactive")) merge_function() else merge_function
+        data <- lapply(datasets$datanames(), function(x) datasets$get_data(x, filtered = TRUE))
+        names(data) <- datasets$datanames()
+        join_keys <- datasets$get_join_keys()
         check_merge_function(merge_fun_name)
 
         ds <- Filter(Negate(is.null), lapply(selector_list(), function(x) x()))
         validate(need(length(ds) > 0, "At least one dataset needs to be selected"))
-        merge_datasets(
-          ds,
-          datasets = datasets,
+        merged_data <- merge_datasets(
+          selector_list = ds,
+          data = data,
+          join_keys = join_keys,
           merge_function = merge_fun_name,
           anl_name = anl_name
         )
+        ch <- teal.code::chunks_new()
+        teal.code::chunks_reset(envir = list2env(data), chunks = ch)
+        for (chunk in merged_data$expr) teal.code::chunks_push(expression = chunk, chunks = ch)
+        teal.code::chunks_safe_eval(chunks = ch)
+
+        merged_data$data <- reactive({ch$get("ANL")})
+        merged_data$chunks <- ch
+        merged_data
```

Comment:
merged_data$expr used to be a character concatenated by "\n". Please 
```suggestion
        merged_data$expr <- paste(merged_data$expr, collapse = "\n")
        merged_data
```
```suggestion
        merged_data
```